File: targets/go/gin/query_test.go
package main

import (
	"testing"
)

// query: parsing, percent-decoding and coercing query parameters, at one and at eight.
//
// The answer is the small payload with an echo of every value the target bound, and the
// values are drawn once per process, so the floor holds each of them as well as the status.
// A target that drops a parameter, coerces one wrong or leaves the %20 in q undecoded
// answers a different echo.

// rb:test query.one
func TestOneQueryParameterIsRead(t *testing.T) {
	a := planFor("query.one")

	got := sendPlanned(a)

	assertFloor(t, a, got)
}

// rb:test query.many
func TestEightOfThemAreReadAndCoerced(t *testing.T) {
	a := planFor("query.many")

	got := sendPlanned(a)

	assertFloor(t, a, got)
}

File: targets/node/h3/suite/query.test.js
/**
 * query: parsing, percent-decoding and coercing query parameters, at one and at eight.
 *
 * The answer is the small payload with the converted values echoed beside it. The plan
 * reader puts the values it drew into both the path and the pinned body, so the floor check
 * is an echo check: a parameter the target dropped, did not coerce or did not decode differs
 * from what was sent. q holds a space sent as %20, which only a percent-decoder turns back
 * into a space.
 */
import { test } from "node:test";
import * as floor from "./floor.js";
import * as planned from "./planned.js";
import { send } from "./fetch.js";

// rb:test query.one
test("one query parameter is read", async () => {
  const a = planned.ask("query.one");

  const answer = await send(a);

  floor.check(a, answer);
});

// rb:test query.many
test("eight of them are read and coerced", async () => {
  const a = planned.ask("query.many");

  const answer = await send(a);

  floor.check(a, answer);
});
